wallet/reference/non-evm-apis/starknet-snap-api.md

```diff
@@ -0,0 +1,1050 @@
+  [BIP-44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki).
+- `deploy`: `boolean` - (Optional) Indicate whether to include send the deploy transaction for the
+  account contract.
+  The default is `false`.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Sepolia testnet.
+
+#### Returns
+
+The address of the account and the transaction hash if the account has been created.
+
+#### Example
+
+<Tabs>
+<TabItem value="Request">
+
+```js
+await getEip6963Provider.request({
+  method: "wallet_invokeSnap",
+  params: {
+    snapId: "npm:@consensys/starknet-snap",
+    request: {
+      method: "starkNet_createAccount",
+      params: {
+        addressIndex: 1,
+        deploy: true,
+        chainId: "0x534e5f5345504f4c4941"
+      },
+    },
+  },
+})
+```
+
+</TabItem>
+<TabItem value="Result">
+
+```json
+{
+  "transaction_hash": "0x05a56e2d52c817161883f50c441c3228cfe54d9f84b5b5b8b1c8b8e0e6f7e6d8",
+  "address": "0xb60e8dd61c5d32be8058bb8eb970870f07233155"
+}
+```
+
+</TabItem>
+</Tabs>
+
+### `starkNet_declareContract`
+
+Registers a contract's class on the Starknet blockchain without deploying it.
+
+#### Parameters
+
+- `senderAddress`: `string` - Address of the sender.
+- `chainId`: `string` - (Optional) ID of the target Starknet network. The default is the Starknet Sepolia testnet.
+- `contractPayload`: `object` - Transaction payload to be deployed.
+  - `contract`: `CompiledContract` | `string` - The compiled contract code ([in Cairo](https://book.cairo-lang.org/ch13-00-introduction-to-starknet-smart-contracts.html)).
+  - `classHash?`: `string` - (Optional) The computed class hash of compiled contract.
+  - `casm?`: CompiledContract | `string` - (Optional) - The compiled [casm](https://docs.starknet.io/architecture-and-concepts/smart-contracts/cairo-and-sierra/).
+  - `compiledClassHash?`: `string` - (Optional) The compiled class hash from casm.
+- `invocationsDetails`: `object` - (Optional) Transaction details object containing:
+  - `nonce`: (Optional) Nonce for the transaction.
+  - `blockIdentifier`: (Optional) Block identifier for the transaction.
+  - `maxFee`: (Optional) Maximum gas fee allowed for the transaction. If not specified, the fee is automatically calculated.
+  - `tip`: (Optional) Additional fee to incentivize miners.
+  - `paymasterData`: (Optional) Paymaster-related data for the transaction.
+  - `accountDeploymentData`: (Optional) Data for account deployment.
+  - `nonceDataAvailabilityMode`: (Optional) Mode for nonce data availability.
+  - `feeDataAvailabilityMode`: (Optional) Mode for fee data availability.
+  - `version`: (Optional) The transaction version.
+  - `resourceBounds`: (Optional) The boundaries on resource consumption during the transaction.
+  - `skipValidate`: `boolean` - (Optional) Skip validation of the transaction.
+
+### Returns
+
+The confirmation of sending a transaction on the starknet contract
```

Comment:
Can you clarify this? Is the confirmation an object that contains the `transaction_hash` and `class_hash`?

Reply:
Yes, it will returns a object contains both transaction_hash and class_hash

but i also wonder, do we need to mentioned the details response in this API docs

Reply:
@stanleyyconsensys for consistency with the thee docs, we try to provide both the example and results for the dev. Is it okay if we leave in the results so it matches the others docs?